managed/models/database.go

```diff
@@ -869,81 +884,115 @@ type SetupDBParams struct {
 	MigrationVersion *int
 }
 
-// SetupDB runs PostgreSQL database migrations and optionally creates database and adds initial data.
-func SetupDB(sqlDB *sql.DB, params *SetupDBParams) (*reform.DB, error) {
+// SetupDB checks minimal required PostgreSQL version and runs database migrations. Optionally creates database and adds initial data.
+func SetupDB(ctx context.Context, sqlDB *sql.DB, params *SetupDBParams) (*reform.DB, error) {
 	var logger reform.Logger
 	if params.Logf != nil {
 		logger = reform.NewPrintfLogger(params.Logf)
 	}
+
 	db := reform.NewDB(sqlDB, postgresql.Dialect, logger)
+	errCV := checkVersion(ctx, db)
+	if pErr, ok := errCV.(*pq.Error); ok && pErr.Code == "28000" {
+		// invalid_authorization_specification	(see https://www.postgresql.org/docs/current/errcodes-appendix.html)
+		if err := initWithRoot(params); err != nil {
+			return nil, errors.Wrapf(err, "couldn't connect do database with provided credentials: %s; tried to create user and database", errCV)
```

Comment:
Let's do one more thing.
In the case the user-provided credentials are incorrect, we are hiding the actual error message from line. We shall expose it.

Can you double-check what we print out in this case? I think it's going to be an error from `initWithRoot` instead of the `errCV`.

Reply:
`WARN[2023-01-17T11:16:15.559+00:00] Failed to migrate database: couldn't connect to database with provided credentials. Tried to create user and database. Error: pq: pg_hba.conf rejects connection for host "172.30.0.3", user "pmm-managed", database "pmm-managed", no encryption: pq: no pg_hba.conf entry for host "172.30.0.3", user "postgres", database "postgres", no encryption.  component=migration`